src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala

```diff
@@ -253,7 +253,12 @@ abstract class BCodeHelpers extends BCodeIdiomatic {
       def fail(msg: String, pos: Position = sym.pos) = {
         reporter.warning(sym.pos,
           sym.name +
-          s" has a main method with parameter type Array[String], but ${sym.fullName('.')} will not be a runnable program.\n  Reason: $msg"
+          s""" has a main method with parameter type Array, but ${sym.fullName('.')} will not be a runnable program.
```

Comment:
we could pass in the signature and rephrase as `a main method with signature $info, but ...`